src/main/java/org/openstreetmap/atlas/geography/boundary/CountryToShardListCache.java

```diff
@@ -0,0 +1,172 @@
+ *
+ * @author james-gage
+ */
+public final class CountryToShardListCache extends Command
+{
+    private static final Logger logger = LoggerFactory.getLogger(CountryToShardListCache.class);
+    private static final Switch<CountryBoundaryMap> BOUNDARIES = new Switch<>("boundaries",
+            "The country boundaries.", value -> initializeCountryBoundaryMap(value),
+            Optionality.REQUIRED);
+    private static final Switch<StringList> COUNTRIES = new Switch<>("countries",
+            "CSV list of the country iso3 codes", value -> StringList.split(value, ","),
+            Optionality.OPTIONAL);
+    private static final Switch<Sharding> SHARDING = new Switch<>("sharding",
+            "File containing the sharding definition (Works with files only)", Sharding::forString,
+            Optionality.REQUIRED);
+    private static final Switch<File> OUTPUT = new Switch<>("output", "The output file", File::new,
+            Optionality.REQUIRED);
+    private static final String DELIMITER = "||";
+    private final MultiMap<String, SlippyTile> countryToShards = new MultiMap<>();
+
+    public static void main(final String[] args)
+    {
+        new CountryToShardListCache().run(args);
+    }
+
+    private static CountryBoundaryMap initializeCountryBoundaryMap(final String value)
+    {
+        final Time start = Time.now();
+        logger.info("Loading boundaries");
+        final CountryBoundaryMap result = CountryBoundaryMap.fromPlainText(new File(value));
+        logger.info("Loaded boundaries in {}", start.elapsedSince());
+        return result;
+    }
+
+    public CountryToShardListCache(final CountryBoundaryMap boundaries, final Sharding sharding)
+    {
+        this(boundaries, new StringList(boundaries.allCountryNames()), sharding);
+    }
+
+    public CountryToShardListCache(final CountryBoundaryMap boundaries, final StringList countries,
+            final Sharding sharding)
+    {
+        CountryShardListing.countryToShardList(countries, boundaries, sharding)
+                .forEach((country, shardSet) ->
+                {
+                    shardSet.forEach(shard -> this.countryToShards.add(country,
+                            SlippyTile.forName(shard.getName())));
+                });
+    }
+
+    public CountryToShardListCache(final Resource resource)
+    {
+        try
+        {
+            resource.lines().forEach(line ->
+            {
+                final String[] countryAndShardList = line.split(Pattern.quote(DELIMITER));
+                final String country = countryAndShardList[0];
+                final String shardList = countryAndShardList[1];
+                Arrays.asList(shardList.split("\\s*,\\s*")).stream().map(SlippyTile::forName)
+                        .forEach(slippyTile -> this.countryToShards.add(country, slippyTile));
+            });
+        }
+        catch (final Exception e)
+        {
+            logger.error("Error while reading CountryToShardListCache file!", e);
```

Comment:
Throw a CoreException here that references the other one! I don't think we want a partially built file, with an error message buried in the logs..